crates/runtime/src/gc/enabled.rs

```diff
@@ -333,30 +344,92 @@ impl VMExternRef {
             ptr::write(
                 extern_data_ptr,
                 VMExternData {
-                    ref_count: AtomicUsize::new(1),
+                    ref_count: Cell::new(1),
                     // Cast from `*mut T` to `*mut dyn Any` here.
                     value_ptr: SendSyncPtr::new(NonNull::new_unchecked(value_ptr.as_ptr())),
                 },
             );
 
             log::trace!("New externref data @ {:p}", extern_data_ptr);
-            VMExternRef(NonNull::new_unchecked(extern_data_ptr).into())
+            VMExternRef(VMGcRef::from_non_null(NonNull::new_unchecked(
+                extern_data_ptr as *mut u8,
```

Comment:
Whiel you're here, mind removing `as` casts where possible and replacing them with `.cast()`?